imagebuildah/stage_executor.go

```diff
@@ -363,24 +363,78 @@ func (s *StageExecutor) Copy(excludes []string, copies ...imagebuilder.Copy) err
 		preserveOwnership := false
 		contextDir := s.executor.contextDir
 		if len(copy.From) > 0 {
+			// If additionalContext is selected do not process anything else
+			didSelectAdditionalContext := false
+			// If additionalContext is selected but it was image only process
+			// image and ignore stages if any. [ Buildx parity ]
+			didSelectImageAsAdditionalContext := false
 			// If from has an argument within it, resolve it to its
 			// value.  Otherwise just return the value found.
 			from, fromErr := imagebuilder.ProcessWord(copy.From, s.stage.Builder.Arguments())
 			if fromErr != nil {
 				return errors.Wrapf(fromErr, "unable to resolve argument %q", copy.From)
 			}
-			if isStage, err := s.executor.waitForStage(s.ctx, from, s.stages[:s.index]); isStage && err != nil {
-				return err
+			if additionalBuildContext, ok := s.executor.additionalBuildContexts[from]; ok {
+				if additionalBuildContext != nil {
```

Comment:
If we never put `nil` values in the map, then we don't need to check for them.

Reply:
Just added a check so in future if someone edits top level code we don't endup SEGSEV. But I can drop it if needed.

Reply:
Go ahead and drop it.  I can't conceive of a use case for setting `nil` values in this map.

Reply:
dropped this nil and other similar nil check for additionalBuildContext.